Comp2024/src/main/java/frc/robot/subsystems/LED.java

```diff
@@ -0,0 +1,145 @@
+//
+// LED subystem - LED feedback on robot
+//
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.led.CANdle;
+
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.LEDConsts.LEDColor;
+import frc.robot.Constants.Ports;
+
+//
+// LED subsystem class
+//
+public class LED extends SubsystemBase
+{
+  // Member objects
+  private final CANdle                    m_candle        = new CANdle(Ports.kCANID_CANdle);
+
+  private final SendableChooser<LEDColor> m_ledChooser    = new SendableChooser<LEDColor>( );
+  private LEDColor                        m_previousColor = LEDColor.LEDCOLOR_OFF;
+
+  // Constructor
+  public LED( )
+  {
+    setName("LED");
+    setSubsystem("LED");
+
+    setColor(LEDColor.LEDCOLOR_BLUE);
+    m_candle.configBrightnessScalar(0.7);
+
+    // Add options for colors in SmartDashboard
+    m_ledChooser.setDefaultOption("LED_Off", LEDColor.LEDCOLOR_OFF);
+    m_ledChooser.addOption("LED_White", LEDColor.LEDCOLOR_WHITE);
+    m_ledChooser.addOption("LED_Red", LEDColor.LEDCOLOR_RED);
+    m_ledChooser.addOption("LED_Orange", LEDColor.LEDCOLOR_ORANGE);
+    m_ledChooser.addOption("LED_Yellow", LEDColor.LEDCOLOR_YELLOW);
+    m_ledChooser.addOption("LED_Green", LEDColor.LEDCOLOR_GREEN);
+    m_ledChooser.addOption("LED_Blue", LEDColor.LEDCOLOR_BLUE);
+    m_ledChooser.addOption("LED_Purple", LEDColor.LEDCOLOR_PURPLE);
+
+    SmartDashboard.putData("LED_Color", m_ledChooser);
+    SmartDashboard.putBoolean("LED_normalMode", true);
+
+    initialize( );
+  }
+
+  @Override
+  public void periodic( )
+  {
+    // This method will be called once per scheduler run
+    setColor(m_ledChooser.getSelected( ));

```

Comment:
While this does work, it causes this subsystem to grab the dashboard setting and push it to the CANdle every 20 msec. even though the user might never change it. Generally beware of putting things in periodic methods unless they are necessary. The correct behavior would require the user to hit a SmartDashboard button that called this line of code from a command.